js/src/admin/components/ExtensionPage.js

```diff
@@ -0,0 +1,387 @@
+          {app.translator.trans('core.admin.extension.uninstall_button')}
+        </Button>
+      );
+    }
+
+    return items;
+  }
+
+  infoItems() {
+    const items = new ItemList();
+
+    if (this.extension.authors) {
+      let authors = [];
+
+      Object.keys(this.extension.authors).map((author, i) => {
+        const link = this.extension.authors[author].homepage
+          ? this.extension.authors[author].homepage
+          : 'mailto:' + this.extension.authors[author].email;
+
+        authors.push(
+          <Link href={link} external={true} target="_blank">
+            {this.extension.authors[author].name}
+          </Link>
+        );
+
+        if (i < this.extension.authors.length - 1) {
+          authors.push(', ');
+        }
+      });
+
+      items.add('authors', [icon('fas fa-user'), <span>{authors}</span>]);
+    }
+
+    const infoData = {};
+
+    if (this.extension.source || this.extension.support) {
+      infoData.source = {
+        icon: 'fas fa-code',
+        href: this.extension.source ? this.extension.source.url : this.extension.support.source,
+      };
+    }
+
+    Object.keys(this.infoFields).map((field) => {
+      if (this.extension.extra['flarum-extension'].info && this.extension.extra['flarum-extension'].info[field]) {
+        infoData[field] = {
+          icon: this.infoFields[field],
+          href: this.extension.extra['flarum-extension'].info[field],
+        };
+      }
+    });
+
+    Object.entries(infoData).map(([field, value]) => {
+      items.add(
+        field,
+        LinkButton.component(
+          {
+            href: value.href,
+            icon: value.icon,
+            external: true,
+            target: '_blank',
+          },
+          app.translator.trans(`core.admin.extension.info_links.${field}`)
+        )
+      );
+    });
+
+    return items;
+  }
+
+  submitButton() {
+    return (
+      <Button onclick={this.saveSettings.bind(this)} className="Button Button--primary" loading={this.loading} disabled={!this.changed()}>
+        {app.translator.trans('core.admin.settings.submit_button')}
+      </Button>
+    );
+  }
+
+  /**
+   * getSettings accepts an array of settings.
+   * Depending on the type of input, you can set the type to 'bool', 'select', or
+   * any standard <input> type.
+   *
+   * @example
+   * [
+   *   {
+   *      setting: 'acme.checkbox',
+   *      label: app.translator.trans('acme.admin.setting_label'),
+   *      type: 'bool'
+   *   }
+   * ]
+   *
+   * @example
+   * [
+   *   {
+   *      setting: 'acme.select',
+   *      label: app.translator.trans('acme.admin.setting_label'),
+   *      type: 'select',
+   *      options: {
+   *        'option1': 'Option 1 label',
+   *        'option2': 'Option 2 label',
+   *      },
+   *      default: 'option1',
+   *   }
+   * ]
+   *
+   * @param settings
+   * @returns {JSX.Element}
+   */
+  getSettings(settings) {
```

Comment:
Ah, one more idea. In `registerSettings`, providing them as an array makes sense. Or actually, maybe that's too much too since a priority can be provided. But here, I think this method would make more sense if called on individual settings (and have the `map` be in the code calling this method).

Reply:
Is there any advantage to having `getSettings` take an array over calling it individually for each setting?

Reply:
Not really. I'm not sure which way looks better.